src/photos/ducks/clustering/albums.js

```diff
@@ -50,17 +68,47 @@ export const findAutoAlbums = async () => {
   ])
   const results = await cozyClient.data.query(autoAlbums, {
     selector: { auto: true },
-    sort: [{ name: 'asc' }]
+    sort: [{ name: 'desc' }]
   })
   return results
 }
 
-export const saveClustering = async (clusters, albums) => {
+const addToClusters = async (clusters, albumsToSave) => {
+  let i = 0
+  for (const photos of clusters) {
+    await addAutoAlbumReferences(photos, albumsToSave[i++])
```

Comment:
Actually, I would inline this function. It really really needs to have the same number of clusters and albums, si it would be good to keep the function close to the condition. This also lessens the risk of someone calling this function from somewhere else, unaware of he requirement.

Note that you can avoid having to carry around an `i` variable by doing `await Promise.all(clusters.map((photos, index) => addAutoAlbumReferences(photos, albumsToSave[index]))`